src/server/pachyderm_test.go

```diff
@@ -5332,39 +5332,48 @@ func TestMaxQueueSize(t *testing.T) {
 			Transform: &pps.Transform{
 				Cmd: []string{"bash"},
 				Stdin: []string{
-					"sleep 10",
+					"sleep 5",
 				},
 			},
 			Input: client.NewAtomInput(dataRepo, "/*"),
 			ParallelismSpec: &pps.ParallelismSpec{
 				Constant: 2,
 			},
 			MaxQueueSize: 1,
+			ChunkSpec: &pps.ChunkSpec{
+				Number: 10,
+			},
 		})
 	require.NoError(t, err)
-	// Get job info 2x/sec for 20s until we confirm two workers for the current job
-	require.NoError(t, backoff.Retry(func() error {
-		jobs, err := c.ListJob(pipeline, nil, nil)
-		if err != nil {
-			return fmt.Errorf("could not list job: %s", err.Error())
-		}
-		if len(jobs) == 0 {
-			return fmt.Errorf("failed to find job")
-		}
-		jobInfo, err := c.InspectJob(jobs[0].Job.ID, false)
-		if err != nil {
-			return fmt.Errorf("could not inspect job: %s", err.Error())
-		}
-		if len(jobInfo.WorkerStatus) != 2 {
-			return fmt.Errorf("incorrect number of statuses: %v", len(jobInfo.WorkerStatus))
-		}
+
+	var jobInfo *pps.JobInfo
+	for i := 0; i < 10; i++ {
+		require.NoError(t, backoff.Retry(func() error {
+			jobs, err := c.ListJob(pipeline, nil, nil)
+			if err != nil {
+				return fmt.Errorf("could not list job: %s", err.Error())
+			}
+			if len(jobs) == 0 {
+				return fmt.Errorf("failed to find job")
+			}
+			jobInfo, err = c.InspectJob(jobs[0].Job.ID, false)
+			if err != nil {
+				return fmt.Errorf("could not inspect job: %s", err.Error())
+			}
+			if len(jobInfo.WorkerStatus) != 2 {
+				return fmt.Errorf("incorrect number of statuses: %v", len(jobInfo.WorkerStatus))
+			}
+			return nil
+		}, backoff.RetryEvery(500*time.Millisecond).For(60*time.Second)))
+
 		for _, status := range jobInfo.WorkerStatus {
 			if status.QueueSize > 1 {
-				return fmt.Errorf("queue size too big")
+				t.Errorf("queue size too big: %d", status.QueueSize)
```

Comment:
It's probably better to make this a `Fatalf` rather than an `Errorf` so that the test exits immediately if this condition is ever met.

Reply:
Sounds good. Will change.